File: tutorials/cpp/tutorial_geometry.cpp
#include "../../src/cpp/core/wafer.hpp"
#include "../../src/cpp/modules/geometry/geometry_manager.hpp"
#include <memory>
#include <iostream>

int main() {
  std::cout << "Tutorial: Geometry Module\n";
  auto wafer = std::make_shared<Wafer>(300.0, 775.0, "silicon");
  GeometryManager geometry;
  std::cout << "Step 1: Initialize grid\n";
  geometry.initializeGrid(wafer, 100, 100);
  std::cout << "Step 2: Apply oxide layer\n";
  geometry.applyLayer(wafer, 0.1, "oxide");
  std::cout << "Final thickness: " << wafer->getThickness() << " um\n";
  return 0;
}

File: tutorials/python/tutorial_geometry.py
from semiconductor_simulator.simulator import Simulator

print("Tutorial: Geometry Module")
simulator = Simulator()
print("Step 1: Initialize grid")
simulator.initialize_geometry(100, 100)
print("Step 2: Apply oxide layer")
simulator.apply_layer(0.1, "oxide")
print("Step 3: Visualize wafer")
simulator.initialize_renderer(400, 400)
simulator.render(simulator.get_wafer())